Q1.java

```diff
@@ -0,0 +1,161 @@
+import java.util.*;
+
+class Q1
+{
+	public static void main(String[] args)
+	{
+		ArrayList<String> words = new ArrayList<String>();
+		words.add("ART");
+		words.add("RAT");
+		words.add("CAT");
+		words.add("CAR");
+
+		printAlphabet(alphabet(words));
+
+	}
+
+
+
+	public static ArrayList<Character> alphabet (ArrayList<String> words)
+	{
+		Map letters = findLetters(words);
+		ArrayList<Node> allNodes = createNodes(letters);
+
+		links(words, allNodes);
+		return getAlphabet(allNodes);
+	}
+
+		
+
+	//getting all of the letters in the language 
+	private static Map findLetters(ArrayList<String> words)
+	{
+		Map letters = new HashMap();
+
+		for(int i=0; i<words.size(); i++)
+		{
+			for(int j=0; j<words.get(i).length(); j++)
+			{
+				if(!letters.containsKey(words.get(i).charAt(j)))
+				{
+					letters.put(words.get(i).charAt(j), false);
+				}				
+			}
+		}
+		return letters;
+	}
+		
+
+
+	//creating all of the nodes 
+	private static ArrayList<Node> createNodes (Map letters)
+	{
+		Set<Character> allLetters = letters.keySet();
+
+		ArrayList<Node> allNodes = new ArrayList<Node>();
+
+		Iterator<Character> it = allLetters.iterator();
```

Comment:
It is much more fluent Java to loop like

for (Character c : allLetters) {
  ...
}